### PR TITLE
Combine pitzer and owens jupyter app

### PR DESCRIPTION
**Pull request by matthu017**
Modify form.js to change form depending on cluster selected
Combine `node_type` in each form.yml

### CODE REVIEW

**Comment by matthu017** [comment]
This PR works for correctly changing the form elements but it gets submitted from `node_type` to `owens_node_type`/`pitzer_node_type`. 
1) We can put the help section on a separate widget and add a 'cluster' attribute to options - this however affects the nicely ordered hierarchy of the yml file.
2) The endpoint that receives the data can use regex to match `*node_type` - this is more of a 'hack' than an actual solution though.

File: form.js [hunk 1]
```diff
@@ -96,13 +99,50 @@ function node_type_change_hander() {
...
+  let cluster_input = $('#batch_connect_session_context_cluster');
+  cluster_input.change(cluster_change_handler);
+}
+
+/**
+ * Update UI when cluster changes; node_type_change_handler must be called again for each cluster
+ */
+function cluster_change_handler(){
+  toggle_node_type_visibility();
+  set_node_type_change_handler();
```
Comment (johrstrom) [inline form.js:115]: You only need to set the change handler once.

**Comment by johrstrom** [comment]
I think we'd want to keep one `node_type` field and just add the data in the attributes to indicate the different cluster and handle a lot more in the javascript. 

It seems like if we did this, we'd need a toggle in the `submit.yml.erb` to look for one or the other too. The `submit.yml.erb` expects just `node_type`.
https://github.com/OSC/bc_osc_jupyter/blob/68d625b8c078474854a75a6ffb5d92f2cee6d9b1/submit.yml.erb#L3

Probably even changing the help depending on the cluster selected too.

So every entry would have fields for both clusters like so. 
```yaml
    options:
      - [
          "any",     "any",
          data-min-ppn-owens: 0,
          data-min-ppn-pitzer: 0,
          data-max-ppn-owens: 40,
          data-max-ppn-pitzer: 48,
          data-can-show-cuda-owens: false,
          data-can-show-cuda-pitzer: false
        ]
```

Or maybe if it's easier/cleaner have one `node_type` with options for both and hide all the options for the different cluster.  So you'd hide anything with `data-cluster: "owens"` when pitzer is selected and vice versa. 

```yaml
     - [
          "gpu",     "gpu",
          data-min-ppn: 0,
          data-max-ppn: 40,
          data-can-show-cuda: true
          data-cluster: "owens",
        ]
```

I think the decision would come down to which solution creates simpler javascript, not only for us, but for other folks when we inevitably share this as an example on how to manage this type of situation.

File: form.js (hunk 1 above)
Comment (matthu017) [reply form.js:115]: The node type id changes to `batch_connect_session_context_owens_node_type` and pitzer_node_type. It has to be called for each cluster.

Comment (johrstrom) [reply form.js:115]: Right, and I guess that ties into the other comment, that there can be only one `node_type` form item. Toggling an event handler's source event is not straight forward. I mean "this handler looks for this element to change" is simple to reason about.

**Comment by ericfranz** [comment]
There are also cases unfortunately where we want to support hiding/showing different form elements based on the selection of another, where the hidden and displayed elements have the same input name. Above is a good example: if we went the route of node_type_pitzer and node_type_owens it seems like for the form submission we would want the input element to be called node_type. 

Like if we could do

```
  owens_node_type:
    name: node_type
    widget: select
    label: "Node type"

  pitzer_node_type:
    name: node_type
    widget: select
```

**Comment by matthu017** [comment]
https://github.com/OSC/bc_osc_jupyter/pull/41 - Lets the user combine data for both clusters under one option. Simpler Javascript but if one cluster has gpus and another doesn't, then errors can be thrown because the option `gpu` is still available
Also the user would need to add `-clusterName` to each of the data elements.
```  
          data-min-ppn-owens: 0,
          data-min-ppn-pitzer: 0,
```
https://github.com/OSC/bc_osc_jupyter/pull/42/ - Adds an extra cluster element for all options. I'd lean towards this one because this removes the edge case in the PR above.
Also this code is compatible with the previous format of form.yml
```
          "gpu",     "gpu",
          data-cluster: "owens",

```

**Comment by johrstrom** [comment]
I'm going to close this because we have the 2 PRs listed above as options.

**Pull request closed without merging**